modules/storage-api/src/main/java/org/apache/ignite/internal/storage/DataStorageManager.java

```diff
@@ -17,84 +17,104 @@
                 .collect(toUnmodifiableMap(
-                        StorageEngine::name,
-                        identity(),
-                        (storageEngine1, storageEngine2) -> {
-                            throw new StorageException(String.format(
-                                    "Duplicate key [key=%s, engines=%s]",
-                                    storageEngine1.name(),
-                                    List.of(storageEngine1, storageEngine2)
-                            ));
-                        }
+                        StorageEngineFactory::name,
+                        engineFactory -> engineFactory.createEngine(clusterConfigRegistry, storagePath)
                 ));
     }
 
+    /**
+     * Constructor, overloads {@link DataStorageManager#DataStorageManager(ConfigurationRegistry, Path, Iterable)} with loading factories
+     * through a {@link ServiceLoader}.
+     */
+    public DataStorageManager(
+            ConfigurationRegistry clusterConfigRegistry,
+            Path storagePath
+    ) {
+        this(clusterConfigRegistry, storagePath, ServiceLoader.load(StorageEngineFactory.class));
+    }
+
     /** {@inheritDoc} */
     @Override
-    public void start() {
+    public void start() throws StorageException {
         engines.values().forEach(StorageEngine::start);
     }
 
     /** {@inheritDoc} */
     @Override
-    public void stop() throws Exception {
-        IgniteUtils.closeAll(engines.values().stream().map(engine -> engine::stop));
+    public void stop() throws StorageException {
+        engines.values().forEach(StorageEngine::stop);
     }
 
     /**
-     * Returns the storage engine factories.
+     * Returns the data storage engine by data storage configuration.
      *
-     * <p>NOTE: It is recommended to override only in tests.
+     * @param config Data storage configuration.
      */
-    protected Iterable<StorageEngineFactory> engineFactories() {
-        return ServiceLoader.load(StorageEngineFactory.class);
+    public @Nullable StorageEngine engine(DataStorageConfiguration config) {
+        return engines.get(config.value().name());
     }
 
     /**
-     * Returns the data storage engine by data storage configuration.
+     * Returns a consumer that will set the default {@link TableConfigurationSchema#dataStorage table data storage} depending on the {@link
+     * StorageEngine engine}.
      *
-     * @param config Data storage configuration.
+     * @param defaultDataStorageView View of {@link TablesConfigurationSchema#defaultDataStorage}. For the case {@link
+     *      DataStorageConfigurationSchema#UNKNOWN_DATA_STORAGE} and there is only one engine, then it will be the default, otherwise there
+     *      will be no default.
      */
-    public @Nullable StorageEngine engine(DataStorageConfiguration config) {
-        return engines.get(config.value().name());
+    public Consumer<DataStorageChange> defaultTableDataStorageConsumer(String defaultDataStorageView) {
```

Comment:
Ok, if the only purpose of the consumer is to be invoked, then why would you create it in the first place? Just pass a "change" object here as an argument